libs/shared/console-shared/src/lib/cluster-settings/feature/create-edit-credentials-modal-feature/create-edit-credentials-modal-feature.tsx

```diff
@@ -0,0 +1,109 @@
+import { CloudProviderEnum, ClusterCredentials } from 'qovery-typescript-axios'
+import { useState } from 'react'
+import { FieldValues, FormProvider, useForm } from 'react-hook-form'
+import { useDispatch } from 'react-redux'
+import { editCredentials, postCredentials } from '@qovery/domains/organization'
+import { useModal } from '@qovery/shared/ui'
+import { AppDispatch } from '@qovery/store'
+import CreateEditCredentialsModal from '../../ui/create-edit-credentials-modal/create-edit-credentials-modal'
+
+export interface CreateEditCredentialsModalFeatureProps {
+  onClose: () => void
+  cloudProvider: CloudProviderEnum
+  organizationId: string
+  currentCredential?: ClusterCredentials
+}
+
+export const handleSubmit = (data: FieldValues, cloudProvider: CloudProviderEnum) => {
+  const currentData = {
+    name: data['name'],
+  }
+
+  if (cloudProvider === CloudProviderEnum.AWS) {
+    return {
+      ...currentData,
+      ...{
```

Comment:
here also you don't need to do that, could just add access_key_id: data['access_key_id'],
secret_access_key: data['secret_access_key'], following ...currentData,